### PR TITLE
[Mac] Fix NSImageCacheException exception when getting the app icon

### PR DESCRIPTION
**Pull request by sevoku**
SharedApplication.ApplicationIconImage is not meant to be used
to retrieve the actual bundle icon, but only to change the default icon.

Instead of SharedApplication.ApplicationIconImage get the icon
directly from the bundle by default and fall-back to BrandingService
if not running from an app bundle.

Fixes VSTS #905844

### CODE REVIEW

**Comment by sevoku** [comment]
@monojenkins backport release-8.2

File: main/src/addins/MacPlatform/MacPlatform.cs [hunk 1]
```diff
@@ -765,6 +765,22 @@ void GlobalSetup ()
...
+				if (applicationIcon == null) {
+					// use the bundle icon by default
+					// if run from a bundle, this will be the icon from the bundle icon file.
+					// if not run from a bundle, this will be the default file icon of the mono framework folder.
+					applicationIcon = NSWorkspace.SharedWorkspace.IconForFile (NSBundle.MainBundle.BundlePath);
+				}
+				return applicationIcon;
+			}
+			set {
+				applicationIcon = value;
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MacPlatform.cs]: Should this set `NSApplication.SharedApplication.ApplicationIconImage`?

Comment (sevoku) [reply main/src/addins/MacPlatform/MacPlatform.cs]: Yeah, better logic: 091d8c8

**Comment by sevoku** [comment]
@monojenkins backport release-8.2

**Pull request merged**